hkm/views/views.py

```diff
@@ -1096,23 +1091,34 @@ def handle_download(self, request, *args, **kwargs):
                   'data': {'url': tmp_image.edited_image.url}})
         return http.JsonResponse({'url': tmp_image.edited_image.url})
 
+
+class AjaxAddToCollection(View):
+    name = 'hkm_add_to_collection'
+
+    def post(self, request, *args, **kwargs):
+        action = request.POST['action']
+        if action == 'add':
+            return self.handle_add_to_collection(request, *args, **kwargs)
+        elif action == 'add-create-collection':
+            return self.handle_add_to_new_collection(request, *args, **kwargs)
+        return http.HttpResponseBadRequest()
```

Comment:
Is the requirement of an authenticated user for these actions removed intentionally?

Reply:
The button is hidden in the frontend if user is not authenticated, so I thought that would be enough. Should I add it to backend as well?